pkg/cluster-diagnostics/nodesdebug/nodes_debug.go

```
@@ -0,0 +1,347 @@
+func (n *NodeDebug) pollForRunning(ctx context.Context, namespace string, podName string, timeout time.Duration) error {
+	// Poll until the pod is running or the timeout is reached.
+	err := wait.PollUntilContextTimeout(ctx, 2*time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+		pod, err := n.Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("failed to get debug pod %s status: %w", podName, err)
+		}
+
+		switch pod.Status.Phase {
+		case corev1.PodFailed:
+			// Exit early if pod has failed.
+			return false, fmt.Errorf("debug pod %s failed: %w", podName, pod.Status.Message)
```

golangci-lint (2.12.2)

[error] printf: fmt.Errorf format %w has arg pod.Status.Message of wrong type string

(govet)